zetaclient/orchestrator/bootstrap.go

```diff
@@ -0,0 +1,414 @@
+
+		presentChainIDs = append(presentChainIDs, chainID)
+
+		// noop
+		if mapHas(signers, chainID) {
+			continue
+		}
+
+		cfg, _ := app.Config().GetBTCConfig()
+
+		utxoSigner, err := btcsigner.NewSigner(btcChain, tss, ts, logger, cfg)
+		if err != nil {
+			logger.Std.Error().Err(err).Msgf("Unable to construct signer for UTXO chain %d", chainID)
+			continue
+		}
+
+		addSigner(chainID, utxoSigner)
+	}
+
+	// Remove all disabled signers
+	mapDeleteMissingKeys(signers, presentChainIDs, onBeforeRemove)
+
+	return added, removed, nil
+}
+
+// CreateChainObserverMap creates a map of interfaces.ChainObserver (by chainID) for all chains in the config.
+// - Note (!) that it calls observer.Start() on creation
+// - Note that data race handling is the responsibility of the caller.
+func CreateChainObserverMap(
+	ctx context.Context,
+	client interfaces.ZetacoreClient,
+	tss interfaces.TSSSigner,
+	dbpath string,
+	logger base.Logger,
+	ts *metrics.TelemetryServer,
+) (map[int64]interfaces.ChainObserver, error) {
+	observerMap := make(map[int64]interfaces.ChainObserver)
+
+	_, _, err := syncObserverMap(ctx, client, tss, dbpath, logger, ts, &observerMap)
+	if err != nil {
+		return nil, err
+	}
+
+	return observerMap, nil
+}
+
+// syncObserverMap synchronizes the given observer map with the observers for all chains in the config.
+// This semantic is used to allow dynamic updates to the map.
+// Note (!) that it calls observer.Start() on creation and observer.Stop() on deletion.
+func syncObserverMap(
+	ctx context.Context,
+	client interfaces.ZetacoreClient,
+	tss interfaces.TSSSigner,
+	dbpath string,
+	logger base.Logger,
+	ts *metrics.TelemetryServer,
+	observerMap *map[int64]interfaces.ChainObserver,
+) (int, int, error) {
+	app, err := zctx.FromContext(ctx)
+	if err != nil {
+		return 0, 0, errors.Wrapf(err, "failed to get app context")
+	}
+
+	var (
+		added, removed int
+
+		presentChainIDs = make([]int64, 0)
+
+		onAfterAdd = func(_ int64, ob interfaces.ChainObserver) {
+			ob.Start(ctx)
+			added++
+		}
+
+		addObserver = func(chainID int64, ob interfaces.ChainObserver) {
+			mapSet[int64, interfaces.ChainObserver](observerMap, chainID, ob, onAfterAdd)
+		}
+
+		onBeforeRemove = func(_ int64, ob interfaces.ChainObserver) {
+			ob.Stop()
+			removed++
+		}
+	)
+
+	// EVM observers
+	for _, evmConfig := range app.Config().GetAllEVMConfigs() {
+		var (
+			chainID   = evmConfig.Chain.ChainId
+			chainName = evmConfig.Chain.ChainName.String()
+		)
+
+		if evmConfig.Chain.IsZetaChain() {
+			continue
+		}
+
+		chainParams, found := app.GetEVMChainParams(evmConfig.Chain.ChainId)
+		switch {
+		case !found:
+			logger.Std.Error().Msgf("Unable to find chain params for EVM chain %d", chainID)
+			continue
+		case !chainParams.IsSupported:
+			logger.Std.Error().Msgf("EVM chain %d is not supported", chainID)
+			continue
+		}
+
+		presentChainIDs = append(presentChainIDs, chainID)
+
+		// noop
+		if mapHas(observerMap, chainID) {
```

Comment:
As mentioned earlier in Slack. For incoming refactor, once getting rid of `GetAllEVMConfigs()`, `GetBTCChainAndConfig` etc., we don't need to check `mapHas` any more as there will be only one single map in the app context exposing two methods:
1. `GetEnabledBTCChains() []chains.Chain`
2. `GetExternalChainParams(chainID int64)`

we'll simply iterate all supported chains and get their corresponding chain params by a chain ID. You can still refer to the implementation [here](https://github.com/zeta-chain/node/blob/orchestrator-add-new-chain-at-runtime/zetaclient/context/app_context.go#L123)